### ISSUE TITLE
Missing temperature data?

### ISSUE THREAD

**Issue by anferico** [Lucky-Loek/ieee-phm-2012-data-challenge-dataset#2]
I've noticed that there are no temperature data inside the `Learning_set` directory. Such data are missing in the following subdirectories as well:
- `Test_set/Bearing1_3`
- `Test_set/Bearing2_3`
- `Test_set/Bearing2_6`

Other subdirectories inside `Test_set` do contain temperature data. 

Are they meant to be missing or is this dataset incomplete?

**Comment by Lucky-Loek** [comment]
This is a literal copy of the dataset that FEMTO-ST provided. So for any questions, please ask the cited authors.

I think it is by design. The .pdf file states that the authors toyed around with the dataset a little so it would be harder to come to good solutions. I think one of those measures is removing temperature data in the test set.

**Comment by anferico** [comment]
> I think one of those measures is removing temperature data in the test set.

Yeah, I guess that'd be reasonable, but if there are no temperature data in the **training set**, how am I supposed to make any use of them during the testing phase (when they're present)?

**Comment by Lucky-Loek** [comment]
> Yeah, I guess that'd be reasonable, but if there are no temperature data in the **training set**, how am I supposed to make any use of them during the testing phase (when they're present)?

That's what the authors want you to figure out I guess? Again, contact them. This repo is just a mirror of the dataset since the official link of the Data Challenge is dead.

But to give a little hint: if temperature suddenly spikes but the vibration data doesn't show anything yet... Might that be a descriptor of things happening? Or maybe the other way around?

**Comment by anferico** [comment]
Sure I will contact them, I was just trying to discuss that problem with you. I suppose that if I really want to use temperature data in the testing phase, I'd have to augment the training set with some dummy temperature data, otherwise there would be an inconsistency in that the model was trained with data having two features (horizontal and vertical accelerations), but at test time data have one more feature (temperature).

On top of that, while doing this, I can't look at temperature data present in the test set, as I would introduce a bias in the model

**Comment by Lucky-Loek** [comment]
> I was just trying to discuss that problem with you.

Ah okay, didn't notice that. Sorry for the quick close-comment. My solution was to ignore the temperature data and solely focus on the vibration data. Created a neural network that would eventually score 0.30 IIRC. Then again, I haven't looked at this challenge for two years ;-)

> I'd have to augment the training set with some dummy temperature data, otherwise there would be an inconsistency in that the model was trained with data having two features (horizontal and vertical accelerations), but at test time data have one more feature (temperature).
> On top of that, while doing this, I can't look at temperature data present in the test set, as I would introduce a bias in the model

Interesting points! At the time of this challenge (2012), Machine Learning wasn't really used as much as it is now. It's interesting to look up the papers of the competitors. Pretty much every solution involves some pretty complex physics and statistics to come to a solution. I haven't spotted one that actually uses machine learning. I guess the temperature problem only arises whenever you try to make a "black box" solution like machine learning does.

**Comment by anferico** [comment]
> Interesting points! At the time of this challenge (2012), Machine Learning wasn't really used as much as it is now. It's interesting to look up the papers of the competitors. Pretty much every solution involves some pretty complex physics and statistics to come to a solution. I haven't spotted one that actually uses machine learning. I guess the temperature problem only arises whenever you try to make a "black box" solution like machine learning does.

Well, that makes much more sense now. For some reason, I was assuming that you were meant to tackle this problem using machine learning. Alright then, thanks for your reply ;-)

**Comment by anferico** [comment]
For your interest, I've found another version of this dataset in [NASA's Prognostic Data Repository](https://ti.arc.nasa.gov/tech/dash/groups/pcoe/prognostic-data-repository/#femto).

Compared to the dataset hosted in this repository, the one that I've referenced _does_ include some temperature data in the training set, namely inside the following subfolders:
- `Learning_set/Bearing1_1`
- `Learning_set/Bearing1_2`
- `Learning_set/Bearing2_1`
- `Learning_set/Bearing3_1`

Note that this is consistent with the information reported in the table found at page 10 of the PDF (section A.4) containing the challenge's details. 

The only inconsistency is that although the table within section A.5 states that the subfolder `Test_set/Bearing2_7` contains vibration data only, it actually contains temperature data, too.

**Comment by Lucky-Loek** [comment]
That's weird? Then there's different data sets of the same challenge? Or maybe I accidentally removed the temperature files because I didn't use them in my neural network... I'll diff the set provided by NASA with mine and see if there's anymore differences when I have some time. 

Good catch, nice!

**Comment by Lucky-Loek** [comment]
@anferico Fixed in the latest commit. Thanks for the research, my repository was indeed missing the temperature data in the learning set.
